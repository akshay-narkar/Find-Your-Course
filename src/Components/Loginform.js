import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

export default function Loginform(props) {
  const { clickHandler, error } = props;

  return (
    <>
      <div className="backgroundsign row align-items-center m-0">
        <div className="container text-center">
          <h1 className="my-2">Welcome Back!</h1>
          <h3 className="my-4">{error}</h3>
          <form onSubmit={clickHandler}>
            <div className="mb-3">
              <label htmlFor="Email" className="form-label fw-bold">
                Email address
                <input type="email" className="form-control fw-bold" id="Email1" required />
              </label>
            </div>
            <div className="mb-3">
              <label htmlFor="Password" className="form-label fw-bold">
                Password
                <input type="password" className="form-control fw-bold" id="Password1" required />
              </label>
            </div>
            <button type="submit" className="btn signing shadow fw-bold">Login</button>
          </form>
          <div className="mt-3">
            <p className="mb-0">New User? Sign Up Here</p>
            <Link to={{ pathname: '/signup' }} className="nav-link active mt-1 pt-1">Sign Up</Link>
          </div>
        </div>
      </div>
    </>
  );
}

Loginform.propTypes = {
  clickHandler: PropTypes.func.isRequired,
  error: PropTypes.string,
};

Loginform.defaultProps = {
  error: '',
};
