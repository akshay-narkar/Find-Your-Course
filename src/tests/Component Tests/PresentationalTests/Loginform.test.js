/*eslint-disable*/
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import Loginform from '../../../Components/Loginform';
import store from '../store';

const error = 'Random error';
const clickHandler1 = () => {};

it('UI is unaltered wrt first snapshot', () => {
  const elem = renderer.create(<BrowserRouter><Provider store={store}><Loginform error={error} clickHandler={clickHandler1} /></Provider></BrowserRouter>).toJSON();

  expect(elem).toMatchSnapshot();
});
