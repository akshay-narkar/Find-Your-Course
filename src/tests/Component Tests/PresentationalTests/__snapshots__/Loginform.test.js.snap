// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UI is unaltered wrt first snapshot 1`] = `
<div
  className="backgroundsign row align-items-center m-0"
>
  <div
    className="container text-center"
  >
    <h1
      className="my-2"
    >
      Welcome Back!
    </h1>
    <h3
      className="my-4"
    />
    <form
      onSubmit={[Function]}
    >
      <div
        className="mb-3"
      >
        <label
          className="form-label fw-bold"
          htmlFor="Email"
        >
          Email address
          <input
            className="form-control fw-bold"
            id="Email1"
            required={true}
            type="email"
          />
        </label>
      </div>
      <div
        className="mb-3"
      >
        <label
          className="form-label fw-bold"
          htmlFor="Password"
        >
          Password
          <input
            className="form-control fw-bold"
            id="Password1"
            required={true}
            type="password"
          />
        </label>
      </div>
      <button
        className="btn signing shadow fw-bold"
        type="submit"
      >
        Login
      </button>
    </form>
    <div
      className="mt-3"
    >
      <p
        className="mb-0"
      >
        New User? Sign Up Here
      </p>
      <a
        className="nav-link active mt-1 pt-1"
        href="/signup"
        onClick={[Function]}
      >
        Sign Up
      </a>
    </div>
  </div>
</div>
`;
